Documentation/CRDs/Cluster/external-cluster/topology-for-external-mode.md

```diff
@@ -0,0 +1,121 @@
+# Topology-Based Provisioning
+
+## Scenario
+Applications like Kafka will have a deployment with multiple running instances. Each service instance will create a new claim and is expected to be located in a different zone. Since the application has its own redundant instances, there is no requirement for redundancy at the data layer. A storage class is created that will provision storage from replica 1 Ceph pools that are located in each of the separate zones.
+
+## Configuration Flags
+
+Add the required flags to the script: `create-external-cluster-resources.py`:
+`--topology-pools`: (optional) Comma-separated list of topology-constrained rbd pools
+`--topology-failure-domain-label`: (optional) K8s cluster failure domain label (example: zone, rack, or host) for the topology-pools that match the ceph domain
+ `--topology-failure-domain-values`: (optional) Comma-separated list of the k8s cluster failure domain values corresponding to each of the pools in the `topology-pools` list
+
+The import script will then create a new storage class named `ceph-rbd-topology`.
+
+## Example Configuration
+
+### Ceph cluster
+
+Looking at the osd tree, before jumping to the steps
+```
+ID  CLASS  WEIGHT   TYPE NAME       STATUS  REWEIGHT  PRI-AFF
+-1         2.00000  root default
+-9         0.50000      host osd-0
+ 3    hdd  0.25000          osd.3       up   1.00000  1.00000
+```
+
+Create the zone-specific CRUSH rule, example:
+crush_rule 5
+```
+ {
+            "rule_id": 5,
+            "rule_name": "rule_host-zone-a-hdd",
+            "type": 1,
+            "steps": [
+                {
+                    "op": "take",
+                    "item": -10,
+                    "item_name": "zone-a~hdd"
+                },
+                {
+                    "op": "choose_firstn",
+                    "num": 0,
+                    "type": "osd"
+                },
+                {
+                    "op": "emit"
+                }
+            ]
+}
+```
+
+Create a replica-1 pool in each failure domain (assume zone for this example). Each pool must be created with a CRUSH rule to limit the pool to OSDs in a specific zone.
+
+!!! note
+   Disable the ceph warning for replica-1 pools: `ceph config set global  mon_allow_pool_size_one true`
+
+Set the flags value based on the pools and failure domains.
+
+--topology-pools=pool-a,pool-b,pool-c
+--topology-failure-domain-label=zone
+--topology-failure-domain-values=zone-a,zone-b,zone-c
```

Comment:
These zones aren't determined until after the analyze the nodes in the next paragraph below, right? Seems like these options should just be moved below with the `create-external-cluster-resources.py` command.